modules/internal/ChapelRange.chpl

```diff
@@ -215,12 +215,21 @@ module ChapelRange {
     compilerError("Bounds of '..' must be integers of compatible types, when specified.");
   }
 
-  // Range builders for partially bounded ranges
-  proc chpl_build_partially_bounded_range(param bt: BoundedRangeType, bound: int(?w))
+  // Range builders for low bounded ranges
+  proc chpl_build_low_bounded_range(param bt: BoundedRangeType, bound: int(?w))
     return new range(int(w), bt, false, bound, bound);
-  proc chpl_build_partially_bounded_range(param bt: BoundedRangeType, bound: uint(?w))
+  proc chpl_build_low_bounded_range(param bt: BoundedRangeType, bound: uint(?w))
     return new range(uint(w), bt, false, bound, bound);
-  proc chpl_build_partially_bounded_range(param bt: BoundedRangeType, bound) {
+  proc chpl_build_low_bounded_range(param bt: BoundedRangeType, bound) {
+    compilerError("Bounds of '..' must be integers of compatible types, when specified.");
+  }
+
+  // Range builders for high bounded ranges
+  proc chpl_build_high_bounded_range(param bt: BoundedRangeType, bound: int(?w))
+    return new range(int(w), bt, false, bound, bound);
+  proc chpl_build_high_bounded_range(param bt: BoundedRangeType, bound: uint(?w))
+    return new range(uint(w), bt, false, bound, bound);
```

Comment:
Apologies if we've been through this before, but is there a good motivation to make this two overloads accepting int/uint rather than a single overload accepting integral?  The [u]int(w) could be replaced with bound.type in that case.